control/include/http/check_data.h

```diff
@@ -60,21 +60,26 @@ class CheckData {
     HEADER_REFERER,
   };
   virtual bool FindHeaderByType(HeaderType header_type,
-                                std::string* value) const = 0;
+                                std::string *value) const = 0;
 
   // A generic way to find any HTTP header.
   // This is for custom HTTP headers, such as x-api-key
   // Envoy platform requires "name" to be lower_case.
-  virtual bool FindHeaderByName(const std::string& name,
-                                std::string* value) const = 0;
+  virtual bool FindHeaderByName(const std::string &name,
+                                std::string *value) const = 0;
 
   // Find query parameter by name.
-  virtual bool FindQueryParameter(const std::string& name,
-                                  std::string* value) const = 0;
+  virtual bool FindQueryParameter(const std::string &name,
+                                  std::string *value) const = 0;
 
   // Find Cookie header.
-  virtual bool FindCookie(const std::string& name,
-                          std::string* value) const = 0;
+  virtual bool FindCookie(const std::string &name,
+                          std::string *value) const = 0;
+
+  // Extracts authentication information from header. Returns true on success,
+  // and saves authentication information in |attrs|. Returns false otherwise.
+  virtual bool GetAuthenticationHeader(
```

Comment:
call it
GetJWTPayload()

// If the request has a JWT token and it is verified, get its payload as string map.

Reply:
s/attrs/payload/

Reply:
Done. Please take a look.